backend/interf.ml

```diff
@@ -17,33 +17,35 @@
    Annotate pseudoregs with interference lists and preference lists. *)
 
 module IntPairSet =
-  Set.Make(struct
+  Hashtbl.MakeSeeded(struct
     type t = int * int
-    let compare ((a1,b1) : t) (a2,b2) =
-      match compare a1 a2 with
-        | 0 -> compare b1 b2
-        | c -> c
+
+    let equal ((a1, b1) : int * int) (a2, b2) =
+      a1 = a2 && b1 = b2
+
+    let hash = Hashtbl.seeded_hash
   end)
 
+let mat = IntPairSet.create 42
```

Comment:
That seems pretty low.

Reply:
Should it be based on Reg.num_registers ?

Reply:
Possibly; I am fairly confident the present value is too low
for the pathological cases. Also, I don't know if it is cargo
cult or rooted in theory, perhaps a prime.

Reply:
`42` is fine here, as Hashtbl dynamically increases the size of the table to keep the load factor reasonable.

Primes are indeed a cargo cult. Their advantage was that they did a better job when certain particularly bad hash functions were in use, but the better approach is to avoid such hashes. Hashtbl rounds up the size to a power of two anyway, so the primality of the initial size has no effect.